### ISSUE TITLE
Handle attach

### ISSUE THREAD

**Issue by rom1504** [mhsjlw/flying-squid#139]
It works really well surprisingly by just using the attach_entity packet, see this : 
![mc-carry](http://download.rom1504.fr/stuff/mc-carry.png)

I started it in 04ab50b7d8f2efcfce4638f60861606e56a734c7

it should be put in a proper function and used when vanilla uses it (boat and all)

**Comment by 109C** [comment]
For any future person implementing this: make sure to update the riding entity's position when the ridee moves.

**Comment by demipixel** [comment]
Adding to @109C , believe you don't need to do this if a player because they'll keep track of their own position based off the entity they're attached to (correct?)

**Comment by rom1504** [comment]
yeah I think that's correct.

**Comment by 109C** [comment]
@demipixel Yes, the client does... but not the server. If we don't update the riders position, the server will think that the player / riding entity hasn't moved at all.

**Comment by rom1504** [comment]
the vanilla client send the position of the player, and we send it to other players in updatePosition.js

**Comment by 109C** [comment]
Didn't know that, but it's a problem if we have non-player riding entities.

**Comment by rom1504** [comment]
yes
